### DISCUSSION TITLE
svg image with xlink:href not including external file reference

### DISCUSSION THREAD

**Discussion by EDEADLINK** [ImageMagick/ImageMagick#6722] [Contribute]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported
- [X] I have verified that I am using the latest version of ImageMagick

### ImageMagick version

7.0.11-5

### Operating system

Linux

### Operating system, version and so on

Linux 5.11.10-arch1-1 GNU/Linux

### Description

An svg can contain references to an external file using xlink:href or href attributes.
Imagemagick no longer takes these references into account.
If inkscape is installed, it will replace the referenced image with one that says "Linked Image not Found".
Withouth inkscape the following errors occur:
```
(process:12001): librsvg-CRITICAL **: 11:54:40.719: Handle could not read or parse the SVG; did you check for errors during the loading stage?

(process:12001): librsvg-CRITICAL **: 11:54:40.719: Handle could not read or parse the SVG; did you check for errors during the loading stage?
magick: negative or zero image size `ref.svg' @ error/image.c/SetImageExtent/2651.
magick: no decode delegate for this image format `ref.svg' @ error/svg.c/ReadSVGImage/3417.
```

### Steps to Reproduce

1. unzip the example in some directory
2. magick example.svg example.png

Expected behaviour:
resulting png contains purple rectangle from referenced file "linked.svg".
Actual behaviour:
resulting png is blank or when inkscape is used contains "Linked Image Not Found".

[example.zip](https://github.com/ImageMagick/ImageMagick/files/6220977/example.zip)

**Comment by snibgo** [comment]
What version of Inkscape?

Trying to read your example.svg in Inkscape v1.0, Inkscape complains with errors and warnings. To fix this, insert xmlns declarations into your SVG:
```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg
   xmlns="http://www.w3.org/2000/svg" 
   xmlns:xlink="http://www.w3.org/1999/xlink"
   width="210mm"
... etc.
```

**Comment by EDEADLINK** [comment]
I am using v1.0.2, but I stripped a bunch of stuff to craft a more minimal testcase.

Here's what inkscape saves using "plain svg" [example.zip](https://github.com/ImageMagick/ImageMagick/files/6229844/example.zip)

It exhibits the same issue.

When I run "magick -verbose example.svg example.png" it outputs:

```
'inkscape' '/tmp/magick-pZ-zncfCxtxrV33AeROcnmv953MTsVor' --export-filename='/tmp/magick-O4O6VaosJzQReJZl4DSAyWfdiZUke4IZ.png' --export-dpi='96' --export-background='rgb(100%,100%,100%)' --export-background-opacity='0.99999999999900002212' > '/tmp/magick-aFXKEODYMt9pxTM3PC2Emoj5kBtaur7M' 2>&
```

So what I think is happening is that magick copies the svg to a tmp file and converts it there.
The problem is that svg allows references by relative path. Which don't work when you move/rename the files.
